### PR TITLE
Adds the ability to not play admin sounds

### LINKED ISSUE #8230: No way to stop bad admin music
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH

### PR DESCRIPTION
**Pull request by ike709**
## About the PR 
Fixes #8230 

Adds an audio setting for playing admin sounds.

**Screenshots**
Make sure you unmute the audio:

https://user-images.githubusercontent.com/5714543/168900554-402be6c6-47dc-4ac1-aeff-2680a8ec0d8a.mp4

**Changelog**

:cl:
- add: Added an audio setting to disable playing admin music

### CODE REVIEW

File: Content.Client/EscapeMenu/UI/Tabs/AudioTab.xaml.cs [hunk 1]
```diff
@@ -77,6 +79,7 @@ private void OnLobbyMusicCheckToggled(BaseButton.ButtonEventArgs args)
             UpdateChanges();
         }
 
+        private void OnAdminSoundsCheckToggled(BaseButton.ButtonEventArgs args)
```
Comment (ike709) [inline Content.Client/EscapeMenu/UI/Tabs/AudioTab.xaml.cs:82]:
```suggestion
        private void OnAdminSoundsCheckToggled(BaseButton.ButtonEventArgs args)
        {
            UpdateChanges();
        }
        
```

**Comment by ike709** [comment]
> 6 days ago

😩

File: Content.Client/Audio/ClientAdminSoundSystem.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        base.Initialize();
+        SubscribeNetworkEvent<AdminSoundEvent>(PlayAdminSound);
+        _cfg.OnValueChanged(CCVars.AdminSoundsEnabled, ToggleAdminSound, true);
+    }
+
+    public override void Shutdown()
+    {
+        base.Shutdown();
+        _adminAudio.Clear();
+    }
```
Comment (metalgearsloth) [inline Content.Client/Audio/ClientAdminSoundSystem.cs:26]: You should shutdown the streams as this is called when the sim shuts down and not necessarily the game (i.e. if you try to reconnect).

File: Content.Client/Audio/ClientAdminSoundSystem.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeNetworkEvent<AdminSoundEvent>(PlayAdminSound);
+        _cfg.OnValueChanged(CCVars.AdminSoundsEnabled, ToggleAdminSound, true);
+    }
+
+    public override void Shutdown()
+    {
+        base.Shutdown();
```
Comment (ike709) [inline Content.Client/Audio/ClientAdminSoundSystem.cs:24]:
```suggestion
        base.Shutdown();
        foreach (var stream in _adminAudio)
        {
            stream?.Stop();
        }
```

**Comment by ike709** [comment]
😩

File: Content.Client/Audio/ClientAdminSoundSystem.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    private void ToggleAdminSound(bool enabled)
+    {
+        if (enabled) return;
+        foreach (var stream in _adminAudio)
+        {
+            stream?.Stop();
+        }
+        _adminAudio.Clear();
+    }
```
Comment (Zumorica) [inline Content.Client/Audio/ClientAdminSoundSystem.cs:48]: Cache the value you get in `ToggleAdminSound` in the entity system so you don't have to `GetCVar` every time `PlayAdminSound` is called

**Review by Zumorica — APPROVED** [review]
Just a small nitpick, other than that looks good!

File: Content.Client/Audio/ClientAdminSoundSystem.cs [hunk 5]
```diff
@@ -9,14 +9,16 @@ namespace Content.Client.Audio;
...
+    
+    private bool _adminAudioEnabled = true;
     private List<IPlayingAudioStream?> _adminAudio = new(1);
 
     public override void Initialize()
     {
         base.Initialize();
         SubscribeNetworkEvent<AdminSoundEvent>(PlayAdminSound);
         _cfg.OnValueChanged(CCVars.AdminSoundsEnabled, ToggleAdminSound, true);
+        _adminAudioEnabled = _cfg.GetCVar(CCVars.AdminSoundsEnabled);
```
Comment (Zumorica) [inline Content.Client/Audio/ClientAdminSoundSystem.cs:21]:
```suggestion
```
No need to get the CVar here as `OnValueChanged` gets called immediately, thanks to the third argument.

**Pull request merged**